### PR TITLE
Alternatives delta fix

### PR DESCRIPTION
**Pull request by Udumft**
### Description
Fixed a bug where after manually switching to alternative route it's duration delta was nullified.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,7 @@
 * Routes now respect conditional access restrictions that depend on the year. ([#4144](https://github.com/mapbox/mapbox-navigation-ios/pull/4144))
 * Routes now respect conditional access restrictions that apply to [all vehicles](https://wiki.openstreetmap.org/wiki/Key:vehicle). ([#4144](https://github.com/mapbox/mapbox-navigation-ios/pull/4144))
 * Fixed an issue where the `Intersection.outletRoadClasses` and `Intersection.outletMapboxStreetsRoadClass` properties were set to `RoadClasses.restricted` for an entrance to a rest area because of its “[local traffic only](https://wiki.openstreetmap.org/wiki/Tag:access%3Ddestination)” restriction. ([#4144](https://github.com/mapbox/mapbox-navigation-ios/pull/4144))
+* Fixed an issue where after manually switching to alternative route it's duration delta in callout text was nullified. ([#4177](https://github.com/mapbox/mapbox-navigation-ios/pull/4177))
```
Comment (MaximAlien) [inline CHANGELOG.md:54]: IMHO it's not clear what is the consequence of this issue (esp. `text was nullified`): text became invalid in callout or it wasn't shown at all?

**Review by MaximAlien — APPROVED** [review]

**Comment by 1ec5** [comment]
Regarding the [SPM integration test failure](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/9595/workflows/cb440d61-be01-452f-aac4-17cf08823c7f/jobs/87511):

>   'project' dependency on 'https://github.com/pointfreeco/swift-snapshot-testing.git' has an explicit name 'SnapshotTesting' which does not match the name 'swift-snapshot-testing' set for 'https://github.com/pointfreeco/swift-snapshot-testing.git'

This seems to be caused by pointfreeco/swift-snapshot-testing#555, which was released a few days ago in [v1.10.0](https://github.com/pointfreeco/swift-snapshot-testing/releases/tag/1.10.0). We’re pulling in that release automatically as part of our dependency on v1.8.1 up to the next major version. For now, #4142 pinned to v1.9.0, so you can rebase to fix the issue.

File: CHANGELOG.md (hunk 1 above)
Comment (Udumft) [reply CHANGELOG.md]: Updated. Hope it explains the fix better now

**Comment by Udumft** [comment]
Since it is a release branch I cannot just rebase and have to pick changes.

**Pull request merged**